docs/design.md

```diff
@@ -0,0 +1,65 @@
+# Designing ORAS
+
+## Introduction
+
+[ORAS](https://oras.land/) is a tool for working with OCI artifacts, especially ones in the OCI registries. It provides user-friendly interfaces to push and pull artifacts to / from registries as well as options for advanced scenarios.
+
+## Unified Experience
+
+The objective of ORAS is simple as transferring artifacts from one place to another.
+
+In the conventional [client-server model](https://en.wikipedia.org/wiki/Client%E2%80%93server_model), the operation of downloading artifacts from the remote registries is referred to as **pull**, and the operation of uploading artifacts from the remote registry is referred to as **push**.
+
+This model can be generalized by abstracting the client and the server as **targets** so that pull and push can be viewed as **copying** from one target to another (see [**Copy API** oras-project/oras-go#8](https://github.com/oras-project/oras-go/pull/8)). For instances,
+
+- Copy from memory to a remote registry.
+- Copy from a remote registry to a local file folder.
+- Copy from a remote registry to another remote registry.
+- Copy from memory to a local file folder.
+
+### Targets
+
+Generally, a target is a [content-addressable storage (CAS)](https://en.wikipedia.org/wiki/Content-addressable_storage) with tags. All blobs in a CAS are addressed by their [descriptors](https://github.com/opencontainers/image-spec/blob/main/descriptor.md).
+
+To retrieve a blob,
+
+1. Get a descriptor. Optionally, it can be resolved by a tag.
+2. Query the blob with a descriptor.
+
+To store a blob,
+
+1. Store the blob directly in the CAS. A descriptor will be returned.
+2. Optionally, associate the returned descriptor with a tag.
+
+It is worth noting that a target is not equal to a registry.

```

Comment:
In the context of ORAS (which is specifically tied to a registry), do we have scenarios where we tag individual blobs other than manifests and index?

Reply:
No, we don't.